File: Java Barcode Scanner ME/app/src/main/java/de/dimedis/mobileentry/backend/response/BaseResponse.java
package de.dimedis.mobileentry.backend.response;

import android.text.TextUtils;

import com.google.gson.annotations.SerializedName;

public class BaseResponse<T> {
    public static final String RESULT_OK = "ok";
    public static final String RESULT_ERROR = "error";

    @SerializedName("request_id")
    public String requestId;
    public String result;
    public T content;
    public ResponseError error;
    public Throwable throwable;

    public boolean isResultOk() {
        return TextUtils.equals(result, RESULT_OK);
    }

    @Override
    public String toString() {
        return "BaseResponse{" + "requestId='" + requestId + '\'' + ", result='" + result + '\'' + ", content=" + content + ", error=" + error + '}';
    }
}

File: Kotlin Migration/NTMobileEntry/app/src/main/java/de/dimedis/mobileentry/backend/response/BaseResponse.kt
package de.dimedis.mobileentry.backend.response

import android.text.TextUtils
import com.google.gson.annotations.SerializedName

open class BaseResponse<T> {
    @SerializedName("request_id")
    var requestId: String? = null
    var result: String? = null
    var content: T? = null
    var error: ResponseError? = null
    var throwable: Throwable? = null
    val isResultOk: Boolean
        get() = TextUtils.equals(result, RESULT_OK)

    override fun toString(): String {
        return "BaseResponse{" +
                "requestId='" + requestId + '\'' +
                ", result='" + result + '\'' +
                ", content=" + content +
                ", error=" + error +
                '}'
    }

    companion object {
        const val RESULT_OK = "ok"
        const val RESULT_ERROR = "error"
    }
}